### PR TITLE
Unexpected TensorBoard behavior

### PR DESCRIPTION
**Pull request by seungjaeryanlee**
Thank you for the great feedback for my last blocker (#1). I moved the RND network to the agent so that it can be trained. To do so, I created a `RndDqnAgent` class that inherits the `DqnAgent` class. The `_loss()` function has been modified to compute RND loss as well, which is used in `_train()` function as intrinsic reward and to train the RND network.

However, I am noticing a strange behavior for TensorBoard. 

Screenshot from 2019-06-28 18-09-29

1. The value is only updated every 1k steps, and
2. RND loss is not being plotted even though it `summary.scalar()` is invoked.

Do you have any ideas about what could be the cause of the problem?

(Please ignore the changes in `rnd_wrapper.py` and `nest_utils.py`, since they will be reverted.)

### CODE REVIEW

File: tf_agents/agents/dqn/rnddqn_agent.py [hunk 1]
```diff
@@ -0,0 +1,456 @@
...
+      #   weighted by 1/N where N is the batch size, even when some of the
+      #   weights are zero due to boundary transitions. Weighting by 1/K where K
+      #   is the actual number of non-zero weight would artificially increase
+      #   their contribution in the loss. Think about what would happen as
+      #   the number of boundary samples increases.
+      avg_rnd_loss = tf.reduce_mean(input_tensor=rnd_losses)
+
+      with tf.name_scope('Losses/'):
+        tf.compat.v2.summary.scalar(
+            name='loss', data=avg_rnd_loss, step=self.train_step_counter)
```
Comment (oars) [inline tf_agents/agents/dqn/rnddqn_agent.py]:
```suggestion
            name='rnd_loss', data=avg_rnd_loss, step=self.train_step_counter)
```

File: tf_agents/agents/dqn/rnddqn_agent.py [hunk 2]
```diff
@@ -0,0 +1,456 @@
...
+
+    # Compute RND loss
+    # RND loss should be calculated first, since it is used as intrinsic rewards
+    with tf.name_scope('rnd_loss'):
+      rnd_prediction, _ = self._rnd_network(time_steps.observation,
+                                            time_steps.step_type)
+      rnd_target, _ = self._target_rnd_network(time_steps.observation,
+                                               time_steps.step_type)
+
+      rnd_losses = element_wise_squared_loss(rnd_prediction, rnd_target)
```
Comment (oars) [inline tf_agents/agents/dqn/rnddqn_agent.py]:
```suggestion
      rnd_losses = rnd_loss_fn(rnd_prediction, rnd_target)
```

File: tf_agents/agents/dqn/rnddqn_agent.py [hunk 3]
```diff
@@ -0,0 +1,456 @@
...
+          multi_dim_actions=multi_dim_actions)
+
+      next_q_values = self._compute_next_q_values(next_time_steps)
+
+      if self._n_step_update == 1:
+        # Special case for n = 1 to avoid a loss of performance.
+        td_targets = compute_td_targets(
+            next_q_values,
+            # TODO Should reward_scale_factor multiply both intrinsic and extrinsic rewards?
+            rewards=reward_scale_factor * (next_time_steps.reward + avg_rnd_loss),
```
Comment (oars) [inline tf_agents/agents/dqn/rnddqn_agent.py]:
```suggestion
            rewards=reward_scale_factor * (next_time_steps.reward + rnd_losses),
```

File: tf_agents/agents/dqn/rnddqn_agent.py (hunk 1 above)
Comment (seungjaeryanlee) [reply tf_agents/agents/dqn/rnddqn_agent.py]: Screenshot from 2019-06-30 01-13-01

This was indeed the problem! Thank you Oscar

File: tf_agents/agents/dqn/rnddqn_agent.py (hunk 2 above)
Comment (seungjaeryanlee) [reply tf_agents/agents/dqn/rnddqn_agent.py]: Thank you for noticing it!

File: tf_agents/agents/dqn/rnddqn_agent.py (hunk 3 above)
Comment (seungjaeryanlee) [reply tf_agents/agents/dqn/rnddqn_agent.py]: Made the change!

Also, I found out that I was using `element_wise_squared_loss()` for RND, which made `rnd_losses.shape == (64, 100)` and `avg_rnd_loss.shape == ()` (64 is the experience batch size, 100 is the final fc layer size). 

I created a new function `mean_squared_loss()`, so now `rnd_losses.shape == (64,)` and `avg_rnd_loss.shape == ()`.

**Comment by seungjaeryanlee** [comment]
Closing this PR since all feedback has been incorporated and all questions have been resolved. Thank you so much!
